airbyte-integrations/connectors/source-appstore-singer/source_appstore_singer/source.py

```diff
@@ -49,25 +50,36 @@ def check_config(self, logger: AirbyteLogger, config_path: str, config: json) ->
         :return: AirbyteConnectionStatus indicating a Success or Failure
         """
         try:
-            # create request fields for testing
+            # If an app on the appstore does not support subscriptions or sales, it cannot pull the relevant reports.
+            # However, the way the Appstore API expresses this is not via clear error messages. Instead it expresses it by throwing an unrelated
+            # error, in this case "invalid vendor ID". There is no way to distinguish if this error is due to invalid credentials or due to
+            # the account not supporting this kind of report. So to "check connection" we see if any of the reports can be pulled and if so
+            # return success. If no reports can be pulled we display the exception messages generated for all reports and return failure.
             api_fields_to_test = {
-                "subscription_report": {"reportType": "SUBSCRIPTION", "frequency": "DAILY", "reportSubType": "SUMMARY", "version": "1_2"}
+                'subscription_event_report': {'reportType': 'SUBSCRIPTION_EVENT', 'frequency': 'DAILY', 'reportSubType': 'SUMMARY', 'version': '1_2'},
```

Comment:
this duplicates some of the changes in the tap because we need to capture error messages and be sure why a failure to fetch a report happened, as it affects whether to fail or succeed the check.